src/server.ts

```diff
@@ -17,7 +17,32 @@ const httpServer = http.createServer(app);
 app.use(express.static(path.join(__dirname, "client")));
 
 app.get("/", async (req, res) => {
-  return res.sendFile(path.join(__dirname, "client", "join.html"));
+  return res.sendFile(path.join(__dirname, "client", "auth.html"));
+});
+
+app.get("/join", async (req, res) => {
+    const token = req.query.token;
+    if (!token) {
+      return res.redirect("/");
+    }
+    return res.sendFile(path.join(__dirname, "client", "join.html"));
+    /*
```

Comment:
I'm assuming this is meant to be commented out...

Reply:
No, I think its correct. If the token param isn't present then redirect to homepage, else forward to the create/join page